winrm/client.go

```diff
@@ -18,10 +18,16 @@ type Client struct {
 	password  string
 	useHTTPS  bool
 	url       string
+	err       error
 	http      HttpPost
 	transport http.RoundTripper
 }
 
+type WinRMError struct {
```

Comment:
This should implement the **Error interface** rather than the **Client**.
Check the last comment.